File: app/src/main/java/com/udacity/maluleque/meutako/MainActivity.java
package com.udacity.maluleque.meutako;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.tabs.TabLayout;
import com.udacity.maluleque.meutako.adapters.FragmentAdapter;
import com.udacity.maluleque.meutako.utils.DateUtils;
import com.udacity.maluleque.meutako.utils.NetworkNotifier;
import com.udacity.maluleque.meutako.utils.NetworkServiceChecker;

import butterknife.BindView;
import butterknife.ButterKnife;


public class MainActivity extends AppCompatActivity implements TransactionListFragment.FabButtonVisibilityListener, NetworkNotifier {

    private static final String TAG = "MainActivity";
    private static final int TRANSACTION_LIST = 0;
    @BindView(R.id.fab)
    FloatingActionButton fab;
    @BindView(R.id.toolbar)
    Toolbar toolbar;
    @BindView(R.id.tabLayout)
    TabLayout tabLayout;
    @BindView(R.id.viewPager)
    ViewPager viewPager;
    private FragmentAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        toolbar.setTitle(R.string.app_name);
        setSupportActionBar(toolbar);

        fab.setOnClickListener(view -> {
            Intent intent = new Intent(this, AddTransactionActivity.class);
            startActivity(intent);
        });

        NetworkServiceChecker networkServiceChecker = new NetworkServiceChecker(this, this);
        networkServiceChecker.execute();

        adapter = new FragmentAdapter(getSupportFragmentManager(), DateUtils.generateDates(), TRANSACTION_LIST);
        viewPager.setAdapter(adapter);
        viewPager.setOffscreenPageLimit(3);
        viewPager.setCurrentItem(10);
        tabLayout.setTabMode(TabLayout.MODE_SCROLLABLE);
        tabLayout.setupWithViewPager(viewPager);


    }



    @Override
    public void hideFabButton() {
        fab.hide();
    }

    @Override
    public void showFabButton() {
        fab.show();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_settings:
                startActivity(new Intent(this, SettingsActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    public void notifyInternetConnection(boolean hasConnection) {
        if (!hasConnection) {
            Toast.makeText(this, "Connect to the internet to backup data", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/udacity/maluleque/meutako/MainActivity.kt
package com.udacity.maluleque.meutako

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.viewpager.widget.ViewPager
import butterknife.BindView
import butterknife.ButterKnife
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.tabs.TabLayout
import com.udacity.maluleque.meutako.TransactionListFragment.FabButtonVisibilityListener
import com.udacity.maluleque.meutako.adapters.FragmentAdapter
import com.udacity.maluleque.meutako.utils.DateUtils
import com.udacity.maluleque.meutako.utils.NetworkNotifier
import com.udacity.maluleque.meutako.utils.NetworkServiceChecker

class MainActivity : AppCompatActivity(), FabButtonVisibilityListener, NetworkNotifier {
    @JvmField
    @BindView(R.id.fab)
    var fab: FloatingActionButton? = null

    @JvmField
    @BindView(R.id.toolbar)
    var toolbar: Toolbar? = null

    @JvmField
    @BindView(R.id.tabLayout)
    var tabLayout: TabLayout? = null

    @JvmField
    @BindView(R.id.viewPager)
    var viewPager: ViewPager? = null
    private var adapter: FragmentAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ButterKnife.bind(this)
        toolbar!!.setTitle(R.string.app_name)
        setSupportActionBar(toolbar)
        fab!!.setOnClickListener { view: View? ->
            val intent = Intent(this, AddTransactionActivity::class.java)
            startActivity(intent)
        }
        val networkServiceChecker = NetworkServiceChecker(this, this)
        networkServiceChecker.execute()
        adapter = FragmentAdapter(supportFragmentManager, DateUtils.generateDates(), TRANSACTION_LIST)
        viewPager!!.adapter = adapter
        viewPager!!.offscreenPageLimit = 3
        viewPager!!.currentItem = 10
        tabLayout!!.tabMode = TabLayout.MODE_SCROLLABLE
        tabLayout!!.setupWithViewPager(viewPager)
    }

    override fun hideFabButton() {
        fab!!.hide()
    }

    override fun showFabButton() {
        fab!!.show()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> {
                startActivity(Intent(this, SettingsActivity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun notifyInternetConnection(hasConnection: Boolean) {
        if (!hasConnection) {
            Toast.makeText(this, "Connect to the internet to backup data", Toast.LENGTH_SHORT).show()
        }
    }

    companion object {
        private const val TAG = "MainActivity"
        private const val TRANSACTION_LIST = 0
    }
}